### ISSUE TITLE
mikt04-sp

### ISSUE THREAD

**Issue by tominatorcz** [nvbach91/4IZ278-2021-2022-LS#102]
# 4FIS events
Autor: Tomáš Mikulenka - mikt04@vse.cz

## Popis aplikace
4FIS events je web na rezervaci/prodej vstupenek na akce studentského spolku 4FIS. Uživatel si vybírá ze seznamu dostupných událostí. Tlačítko na koupi vstupenky vidí uživatel pouze po přihlášení. Po koupi události uživatel vyplní údaje a po potvrzení je. Následně je vygenerován QR kód s lískem na akci a ten zaslán na mail. V případě naplnění kapacity se registrace uzavře. Jelikož je spolek neziskový, vstup na akci je pouze na základě registrace. Akce mohou být spravované administrátorem, ten má právo události zakládat a upravovat. Má také přístup k registrovaným uživatelům.

## Architektura
Webový server apache na eso.vse.cz, na pozadí využítí technologií mysql a php. Na prezentaci front endu HTML, CSS, a javascript s podporou knihovny jquery. Návrhový vzor pro strukturu backendu zatím nemám definovaný.

## Use case diagram
![USE case SP drawio](https://user-images.githubusercontent.com/50962055/164892544-343482af-225e-4943-8bc3-118bfd63409f.png)

## Databázové schéma
**bude doplněno zítra (24.4.2022)

## Wireframe
Wireframe jsem vytvořil na pomezní low/high fidelity pomocí nástroje Figma.

### Hlavní stránka - neregistrovaný uživatel
![Neregistrovany](https://user-images.githubusercontent.com/50962055/164896494-caa9bfc0-2449-4f71-ac59-99e65bdba232.png)

### Hlavní stránka - registrovaný uživatel
![Registrovany](https://user-images.githubusercontent.com/50962055/164896502-df8d5f27-a1e2-462d-80f2-baed6a2cf51d.png)

### Hlavní stránka - administrátor uživatel
![Admin](https://user-images.githubusercontent.com/50962055/164896512-894ebfc3-caa5-4b71-b5d9-91888af19d36.png)

### Registrace uživatele
![Signup](https://user-images.githubusercontent.com/50962055/164896520-ca1a2e47-47ba-4920-8942-60b4d43469f8.png)

### Přihlášení uživatele
![Login](https://user-images.githubusercontent.com/50962055/164896529-d11eb470-1b3f-434d-ac6b-7753e7d92542.png)

### Nákup lístku
![Nakup](https://user-images.githubusercontent.com/50962055/164896537-22a19927-1c7c-4a6e-b831-a516bae8dfc7.png)

### Karta lístky
![Listky](https://user-images.githubusercontent.com/50962055/164896538-ff62a280-bdd0-49ba-951e-214fc6ba9415.png)

### Karta účet
![Ucet](https://user-images.githubusercontent.com/50962055/164896547-5fda8a78-7d17-4c2b-87d1-97d1cb0b40d3.png)

### Karta událost - Administrace události
![Admin - události](https://user-images.githubusercontent.com/50962055/164896551-19fcd101-ae9e-4713-9f96-e9889c593e05.png)

## Doplňující informace
V tomto semestru odevzdávám BP práci a státnicuji. Z důvodu časové optimalizace jsem řešení navrhl tak, abych splnil veškeré požadavky, ale zároveň neimplementuje všechny funkce, které by eshop na prodej lístků měl mít. Obávám se, že bych to zcela jistě nezvládl.

**Comment by nvbach91** [comment]
proc mate samostatnou tabulku lokaci? muze se stat, ze v jedne lokaci budou probihat vice udalosti?

**Comment by tominatorcz** [comment]
> proc mate samostatnou tabulku lokaci? muze se stat, ze v jedne lokaci budou probihat vice udalosti?

Ano, v jedné lokaci může probíhat více událostí. Tabulku jsem zavedl kvůli norm. rozdělení.

**Comment by tominatorcz** [comment]
Tak už chápu problém, narazil jsem při obsluhování události. Při zaklání eventu by byl potřeba i přístup do vytváření lokací. Je to možné, ale pro potřeby projektu zbytečné. Tabulku lokaci tedy ruším. Údaje o lokaci se přesunou do tabulky event.

**Comment by nvbach91** [comment]
super

**Comment by tominatorcz** [comment]
@nvbach91 Dobrý den, chtěl bych se zeptat na podmínku autentizace. Bude stačit ověření v rámci vlastní funkce, nebo je opravdu nutné implementovat oAuth? Nemohu dohledat, na kterém cvičení jsme to probírali.

**Comment by nvbach91** [comment]
Ano je to nutne

**Comment by nvbach91** [comment]
https://github.com/nvbach91/4IZ278-2021-2022-LS/pull/131
